### ISSUE TITLE
$_SERVER var exposes db_name/user/password and salts

### ISSUE THREAD

**Issue by kupoback** [roots/bedrock#474]
* [X] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
* [X] This request isn't a duplicate of an existing issue
* [X] I've read the [docs](https://roots.io/documentation/) and followed them (if applicable)
* [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

## Description

When working on a site, I noticed that if I was to `error_log()` the `$_SERVER` variable that the following additional items would be available (name/user/password removed for bug report):
```
[DB_NAME] => {db_name}
[DB_USER] => {user}
[DB_PASSWORD] => {userspwd}
[WP_ENV] => production
[WP_HOME] => https://domain.test
[WP_SITEURL] => https://wp-boilerplate.test/wp
[AUTH_KEY] => generateme
[SECURE_AUTH_KEY] => generateme
[LOGGED_IN_KEY] => generateme
[NONCE_KEY] => generateme
[AUTH_SALT] => generateme
[SECURE_AUTH_SALT] => generateme
[LOGGED_IN_SALT] => generateme
[NONCE_SALT] => generateme
```

## Steps to reproduce

1. Create a local install, use any environment
2. simply `var_dump()` or `error_log()` the `$_SERVER` var
3. See results

**Expected behavior:** These sensitive data shouldn't be available via that variable

**Actual behavior:** These sensitive data shouldn't be able to output

**Reproduces how often:** 100%

## Versions

Bedrock Install: 1.12.8: 2019-09-05
macOS: 10.14.6
laravel/valet 2.3.3

## Additional information

Basic Sage 9 with Bedrocks install on local.

**Comment by QWp6t** [comment]
Looks like it would be a pretty straightforward change if we care to do this.

```diff
 /**
  * Expose global env() function from oscarotero/env
  */
 Env::init();
 
 /**
  * Use Dotenv to set required environment variables and load .env file in root
  */
-$dotenv = Dotenv\Dotenv::create($root_dir);
+$dotenv = Dotenv\Dotenv::create($root_dir, null, new Dotenv\Environment\DotenvFactory([
+    new Dotenv\Environment\Adapter\PutenvAdapter(),
+]));
 if (file_exists($root_dir . '/.env')) {
     $dotenv->load();
     $dotenv->required(['WP_HOME', 'WP_SITEURL']);
     if (!env('DATABASE_URL')) {
         $dotenv->required(['DB_NAME', 'DB_USER', 'DB_PASSWORD']);
     }
 }
```

https://github.com/vlucas/phpdotenv#loader-customization

**Comment by kupoback** [comment]
Would this also do it for the WP Salts?

**Comment by demyxco** [comment]
+1 can also confirm when running `php -i`, the credentials are also shown.

**Comment by kupoback** [comment]
Tested suggested fix and it removed the `SALTS` from showing as well.

**Comment by petersafwat-flairs** [comment]
Any Updates on this serious issue?

**Comment by austinpray** [comment]
There’s code a couple comments up if you want to test it https://github.com/roots/bedrock/issues/474#issuecomment-544714647

**Comment by kupoback** [comment]
The above does work, but it _should_ be a change merged into the repo for those that are cloning this for projects. This way they don't have to make a note to copy this solution each time.

**Comment by swalkinshaw** [comment]
Well we need a PR with this change 😄 

@QWp6t want to do the honours?

**Comment by montchr** [comment]
@SandiyosDev Before you accuse the maintainers of negligence, I would suggest you ask yourself why you decided that this issue has been solved.  The issue will be closed automatically once the fix is merged.

#598
